openedx_events/learning/data.py

```diff
@@ -0,0 +1,88 @@
+"""
+Data attributes for events within the architecture subdomain `learning`.
+
+These attributes follow the form of attr objects specified in OEP-49 data
+pattern.
+"""
+from typing import Dict
+
+import attr
+from opaque_keys.edx.keys import CourseKey
+
+
+@attr.s(frozen=True)
+class UserProfileData:
+    """
+    Attributes defined for Open edX student's profile object.
+    """
+
+    meta = attr.ib(type=Dict[str, str], factory=dict)
+    name = attr.ib(type=str, factory=str)
+
+
+@attr.s(frozen=True)
+class StudentData:
+    """
+    Attributes defined for Open edX student object.
+    """
+
+    username = attr.ib(type=str)
+    email = attr.ib(type=str)
+    first_name = attr.ib(type=str, factory=str)
+    last_name = attr.ib(type=str, factory=str)
+    is_active = attr.ib(type=bool, default=True)
+    profile = attr.ib(type=UserProfileData, default=UserProfileData())
```

Comment:
Do we definitely want to keep StudentData and UserProfileData as separate entities?
Does it make sense to have one without the other?

This looks like an implementation detail that is just so because of the way edx-platform extends the django auth User. For other packages we could skip this and send it as a single object.

Can you think of a place where we would not want this?

Reply:
Now that I thinking about it, it doesn't make sense to send just the user or just the profile. They always send the whole user.
I'll change it and add a _profile_ namespace.

Besides `meta` and `name`, do you have any other profile fields in mind?